app/routes/armoury/$character/exchange.tsx

```diff
@@ -25,104 +46,218 @@ export async function loader({ request, params }: LoaderArgs) {
-				<div
-					key={offer.id}
-					className={classnames(
-						"border-l-3 relative border-2 border-neutral-400 bg-white p-4 shadow",
-						rarityBorder[String(offer.item.description.overrides.rarity) ?? "0"]
-					)}
-				>
-					<div>{offer.weapon.display_name}</div>
-					<div className="">
-						<div>Rating</div>
-						<span>{offer.item.description.overrides.itemLevel}</span>
-					</div>
-					<div className="">
-						<div>Base Rating</div>
-						<span>{offer.item.description.overrides.baseItemLevel}</span>
-					</div>
-					<div>
-						<div>{offer.item.price.amount.type}</div>
-						<span style={{ color: "gold" }}>
-							{offer.item.price.amount.amount}
-						</span>
-					</div>
-					{offer.item.description.overrides.perks && offer.item.description.overrides.perks.length > 0 ? (
+		<>
+			<div className="grid w-full grow grid-cols-2 gap-4 bg-neutral-200 p-4 shadow-inner">
+				{offers.map((offer) => (
+					<div
+						key={offer.id}
+						className={classnames(
+							"border-l-3 from-1% relative border-2 border-neutral-400 bg-white bg-gradient-to-r shadow",
+							rarityBorder[String(offer.rarity) ?? "0"]
+						)}
+					>
+						<div
+							className={classnames(
+								"m-2 font-bold leading-none",
+								rarityColor[offer.rarity]
+							)}
+						>
+							{offer.shopitem.display_name}
+						</div>
+						<div className="">
+							<span className="m-2 flex items-center font-bold leading-none">
+								<ChevronDoubleUpIcon
+									className="mr-0.5 h-4 w-4"
+									aria-hidden="true"
+								/>
+								<span
+									className={classnames(
+										"m-2 font-bold leading-none",
+										rarityColor[offer.rarity]
+									)}
+								>
+									{offer.item.description.overrides.itemLevel}
+								</span>
+								[{offer.item.description.overrides.baseItemLevel}]
+							</span>
+						</div>
 						<div>
-							<div>Perks</div>
-							{offer.item.description.overrides.perks.map((perk) => (
-								<div key={perk.id}>{perk.id}</div>
-							))}
-							<span></span>
+							<div className="m-2 flex items-center font-bold leading-none text-amber-500">
+								<CircleStackIcon
+									className="mr-0.5 h-4 w-4"
+									aria-hidden="true"
+								/>
+								{offer.item.price.amount.amount}
+							</div>
 						</div>
-					) : null}
+						{offer.item.description.overrides.perks &&
+						offer.item.description.overrides.perks.length > 0 ? (
+							<div>
+								{offer.item.description.overrides.perks.map((perk) => (
+									<div key={perk.id} className="m-2 flex items-center">
+										<Square3Stack3DIcon className="mr-0.5 h-4 w-4" />
+										{perk.id}
+									</div>
+								))}
+								<span></span>
+							</div>
+						) : null}
+
+						<img
+							alt=""
+							loading="lazy"
+							className="max-w-1/2 pointer-events-none absolute right-0 top-0 h-full"
+							src={`https://img.darkti.de/pngs/${offer.shopitem.preview_image}.png`}
+						/>
```

Comment:
check out the Img component I added to the components directory, used in the inventory